### DISCUSSION TITLE
Preferred programs for editing markdown files

### DISCUSSION THREAD

**Discussion by wavingtowaves** [SORTEE-Github-Hackathon/manuscript#23] [General]
Hi everyone, since we are getting started on writing up the manuscript, I'm wondering what programs everyone likes to use for editing markdown files?

I like sublime because it's really lightweight to open and interact with, but there's not really a great "preview" feature that updates with an html version of the markdown file.

I find myself gravitating toward pycharm/rstudio because you can write in markdown and get a nice html preview. 

Could always just edit text right on the files in GitHub, but I like the branching and committing workflow on my own laptop instead of using the web interface.

Thoughts on this? Any good preferences I've missed?

**Comment by colebrookson** [comment]
So I personally really love VS Code for this! I do all my writing of .md files in VS Code now, as they have a really great extension called Markdown Preview Enhanced that does a really nice job of showing you the html preview of the work side by side the .md file. 

Here's what it looks like: 
![Screenshot from 2021-12-18 14-25-29](https://user-images.githubusercontent.com/43683737/146653430-1f6e80f4-39c2-41fc-ae4b-cf0bb7f3e032.png)

Highly recommend, it also does a really nice job of previewing all the formatting as well as code snippets and/or LaTex if you happen to write in that as well :)

**Comment by wavingtowaves** [comment]
Awesome, thanks so much for that suggestion of vscode. A few colleagues use it for sure, but I haven't explored it much at all.

**Comment by pedrohbraga** [comment]
I have heard very good things about Visual Basic. Because I work more often with Rmd documents or R integrations with markdown (_e.g._ that need to include results from R code chunks, and also `bookdown`, `blogdown`), I have been mostly using RStudio.

Whenever the final document is HTML, I use the standard view of the RMarkdown document in combination with `xaringan::infinite_moon_reader()`, `bookdown::serve_book()` or `blogdown::serve_site()`, which shows a live preview of that document in the Viewer pane (which does not require saving to see the outcome!).

If the final document is a PDF (or anything different than HTML), I use the standard view of RMarkdown or the Visual Editor (`Ctrl+Shift+F4`).

The Visual Editor in RStudio is mostly a what-you-see-is-what-you-get tool and it looks like this:

![image](https://user-images.githubusercontent.com/8599229/146718956-261274b8-c9a1-4ef9-9625-6172d2065f69.png)

(Early next year, RStudio will launch a new preview version with a lot of improvements to the Rmd and Md editing!)

**Reply by Aariq** [reply]
You can also tell RStudio visual editor to do automatic text wrapping with one sentence per line! This can be done with a YAML option, that I don't think would affect people not using RStudio. 

```
---
editor_options:
  markdown:
    wrap: sentence
---
```

**Reply by Aariq** [reply]
I just tested out using RStudio's Visual Markdown Editor for this manuscript.  

**The good news:** the YAML header doesn't seem to affect how the manuscript gets built, and project-wide options for the editor are stored in the .Rproj file created when you open the repo as an RStudio project.  

**The bad news:** Visual Markdown Editor mode makes changes to the markdown. Some are innocuous (e.g., changes `__text__` to `**text**`) but other changes break stuff for manubot (e.g. changes `Citation by URL [@{https://greenelab.github.io/meta-review/}].` to `Citation by URL [\@{<https://greenelab.github.io/meta-review/>}].`, which breaks the citation).  

Citing by DOI, and other sorts of standard markdown things are unaffected though.  

I'll go ahead and make a small PR that sets things up for RStudio users.

**Reply by pedrohbraga** [reply]
Hi, @Aariq !

Indeed the bad news is an issue! 

With respect to the citation, it breaks when one switches from the Visual Editor to the Standard Script View, or back to the Visual Editor. 

I raised [this issue](https://github.com/rstudio/rstudio/issues/10075) within RStudio's IDE repository and they have taken care of it! These changes will come into effect in the Prairie Trillium (2022-01.0) version that will come early next year!

About the changes to the editing, do you think that changing the RMarkdown flavour within the YAML could solve that problem (_e.g._  `output: github_document`)?

**Reply by Aariq** [reply]
Oh nice!  I'll keep an eye out for that update.  I'm not sure that anything other than `editor_options:` in YAML has any effect on the visual editor behavior since it even works without a YAML header.  Might be worth trying though.

**Comment by pedrohbraga** [comment]
Another simple alternative that I really like to use on the go is [Brackets](https://brackets.io/), which is a lightweight, open-source, "notepad on steroids". There are tons of extensions, including a Markdown live preview.

![image](https://user-images.githubusercontent.com/8599229/146718445-d9e6d000-e7fa-4f3e-b4bc-28df6832b9dd.png)
